lib/house.rb

```diff
@@ -0,0 +1,47 @@
+class House
+  attr_reader :price, :address, :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+    @above_market_price = "$500,000"
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def add_commas_to_price(price)
+    @price.to_s.reverse.gsub(/(\d{3})(?=\d)/, '\\1,').reverse
+  end
+
+  def above_market_price?
+    @price > @above_market_price
+  end
+
+  def rooms_from_category(category)
+    @rooms.find_all {|room| room.category == category}
+  end
+
+  def area
+    @rooms.sum {|room| room.area}
+  end
+
+  def details(price, address)
```

Comment:
be careful-- your implementation needs to match the interaction pattern exactly.  The way it asks for the method it be called, it takes no arguments.  There really isn't any need for arguments here anyway, since you have access to `@price` and `@address` as instance variables, so there's no reason to pass them in.